55_revocation_actions_without_python.md

```diff
@@ -0,0 +1,311 @@
+notes and implementing the feature itself. Reviewers
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+This enhancement proposal allows revocation actions to be any
+executable or script, not only a Python module.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this enhancement.  Describe why the change is important and the benefits to users.
+-->
+
+Currently Keylime requires that agent-local actions executed at
+revocation notification to be written in Python with a specific
+convention: an async `execute` function is globally defined and it
+takes a revocation message as the argument.
+
+With the upcoming addtion of the new Rust-based keylime agent there
+will need to be a means to execute actions without Python runtime
+installation.
+
+### Goals
+
+<!--
+List the specific goals of the enhancement.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+ * Arbitrary scripts/executable can be used as a revocation action
+
+### Non-Goals
+
+<!--
+What is out of scope for this enhancement?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+ * Secure execution of revocation actions (while it is desired), such as using sandbox and seccomp
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The actions defined in the configuration file (or tenant-provided
+action list) can be any type of script/executable.
+
+When Keylime agent receives a revocation message, it stores the JSON
+payload in a file and invokes revocation actions with the file as a
+command-line argument.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+* The agent is configured with revocation notification enabled
+* The agent is also configured with non-Python actions (pre-installed on the system) set up in the `revocations_actions` configuration option
+* The verifier notifies a revocation
+* The agent will invoke the actions with the revocation message stored in a file
+
+#### Story 2
+
+* The agent is configured with revocation notification enabled
+* The tenant sends non-Python actions as part of the initial payload, as well as the `action_list` file listing those actions
+* The verifier notifies a revocation
+* The agent will invoke the actions with the revocation message stored in a file
+
```

Comment:
If we want to still support the old model (does anyone actually have custom hooks in production?) a user story with a python module would be nice.

Reply:
Agreed. It would be nice to preserve backwards compatibility. This might mean the rust agent has to wrap an older python file in the necessary boiler plate so it can be executed as a standalone script.

Reply:
Expanded the use case to cover the Python actions.